mmv1/third_party/terraform/services/storage/resource_storage_bucket.go.tmpl

```diff
@@ -549,6 +549,22 @@ func ResourceStorageBucket() *schema.Resource {
 					},
 				},
 			},
+			"hierarchical_namespace": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `The bucket's HNS support, which defines bucket can organize folders in logical file system structure`,
+				Elem : &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
```

Comment:
Should we mark this field with ForceNew as well since it is not mutable?

Reply:
currently this block contains only one resource I see no need of adding again here as the whole block is enforce with ForceNew true,

please suggest otherwise I will add ForceNew also at enabled field level @kautikdk

Reply:
So actually, I asked for the ForceNew because users can not update this field. If we don't add this schema behavior then it will start generating diff every time after updating but won't have any effect. I would leave this to you for further investigation and decision.

Reply:
I see, So if the Resource level attributes can directly pass ForceNew behavior to it's nested fields then currently it makes sense to keep it only at top level now but in future if there is requirement, we can remove it from the top level and directly add it to the nested individual fields.

Reply:
This level will also need ForceNew set! I don't think we'd recreate in this case or the inverse as is, only when we removed the block:

```
  hierarchical_namespace = {
    enabled = false
  }
```

to

```
  hierarchical_namespace = {
    enabled = true
  }
```

Reply:
Added ForceNew at enabled field level as well